pkg/controllers/nodeclass/status/controller.go

```diff
@@ -54,6 +54,7 @@ type Controller struct {
 	subnet          *Subnet
 	securitygroup   *SecurityGroup
 	launchtemplate  *LaunchTemplate
+	nodeClassStatus *NodeClassStatus
```

Comment:
```suggestion
	readiness *NodeClassStatus
```

We're already in the nodeclass status controller so I think readiness is enough here. We should also make a `TODO` here to mark that we are going to be removing this eventually when we have sub-status conditions